drivers/vcluster/vcluster.go

```diff
@@ -537,3 +471,156 @@ func (v *VCluster) FetchFIOLogs(podName, namespace string) (string, error) {
 	}
 	return buf.String(), nil
 }
+
+// DeleteJobOnVcluster deletes the Job on Vcluster
+func (v *VCluster) DeleteJobOnVcluster(appNS string, jobName string) error {
+	deletePolicy := metav1.DeletePropagationForeground
+	return v.Clientset.BatchV1().Jobs(appNS).Delete(context.TODO(), jobName, metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy,
+	})
+}
+
+// CreateNginxDeployment Deploys an Nginx Deployment on Vcluster
+func (v *VCluster) CreateNginxDeployment(pvcName string, appNS string, deploymentName string) error {
+	nginxDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deploymentName,
+			Namespace: appNS,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: int32Ptr(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": deploymentName,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": deploymentName,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "nginx-container",
+							Image: "nginx:latest",
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 80,
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									MountPath: "/usr/share/nginx/html",
+									Name:      "nginx-volume",
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "nginx-volume",
+							VolumeSource: corev1.VolumeSource{
+								PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+									ClaimName: pvcName,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	log.Infof("Going ahead to deploy Nginx Application on VCluster %v", v.Name)
+	if _, err := v.Clientset.AppsV1().Deployments(appNS).Create(context.TODO(), nginxDeployment, metav1.CreateOptions{}); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ScaleVclusterDeployment Scales a deployment on VCluster to set replicas
+func (v *VCluster) ScaleVclusterDeployment(appNS string, deploymentName string, replicas int32) error {
+	deployment, err := v.Clientset.AppsV1().Deployments(appNS).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	deployment.Spec.Replicas = &replicas
+	_, err = v.Clientset.AppsV1().Deployments(appNS).Update(context.TODO(), deployment, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+	log.Infof("Successfully scaled Deployment %s to %d replicas.", deploymentName, replicas)
+	return nil
+}
+
+// ValidateDeploymentScaling Validates if a deployment on Vcluster is having expected number of Replicas or not
+func (v *VCluster) ValidateDeploymentScaling(appNS string, deploymentName string, expectedReplicas int32) error {
+	checkDeploymentScaling := func() (interface{}, bool, error) {
+		pods, err := v.Clientset.CoreV1().Pods(appNS).List(context.TODO(), metav1.ListOptions{
```

Comment:
Could this code move to another function called listPods where list pods can be done by providing labels so that it can be leveraged later for listing pods for any other apps?

Reply:
Done